### PR TITLE
linux compatibility

### PR DESCRIPTION
**Pull request by mvexel**
got it to build but not run.

### CODE REVIEW

**Comment by mvexel** [comment]
Hey, it works!

screenshot from 2016-09-11 17-26-15

**Comment by mvexel** [comment]
This was done on Ubuntu 16.04 LTS. I needed to install a bunch of dependencies, see below. Not sure if they are all required.

```
qt5-qmake
qt5-default 
libqt5qml5
libqt5quick5
qtdeclarative5-dev
openssl
libssl-dev 
libcrypto++-dev 
openssh-server
libusb-dev 
libusb++-dev 
libusb-dev 
libusb-1.0-0-dev 
qtcreator
qml-module-qtquick-layouts 
qml-module-qtquick-dialogs 
```

**Comment by mvexel** [comment]
I am getting an authentication problem however, see below (token obfuscated)

```
CLIENT TOKEN "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" USER NAME : "mvexel" USER EXTERNAL ID:  8909
Saved!
Request URL:  "http://openstreetview.com/1.0/sequence/"
Metadata:  ""
"{\"status\":{\"apiCode\":\"401\",\"apiMessage\":\" You have to be authenticated in order to access this method\",\"httpCode\":400,\"httpMessage\":\"Bad Request\"}}"
New Sequence Failed!
Request URL:  "http://openstreetview.com/1.0/sequence/"
Metadata:  ""
"{\"status\":{\"apiCode\":\"401\",\"apiMessage\":\" You have to be authenticated in order to access this method\",\"httpCode\":400,\"httpMessage\":\"Bad Request\"}}"
New Sequence Failed!
pause upload
```

**Comment by Krakonos** [comment]
I did some reseach and found out what's happening. Unfortunately, I have no clue as to why. I suspect there is some memory access problem, but was unable to spot it.

In function ::requestNewSequence a request is created and sent. It uses QHttpMultiPart, which is fine. But sometime during the processing, probably inside m_manager->post(), it causes the headers to be included twice (I changed the url, and deleted parameters, just to rule out other factors):

```
POST / HTTP/1.1
Content-Type: multipart/form-data; boundary="boundary_.oOo._OTI2NTYxNzU=MTk1MzY3MzMwNg==MTY3NjM1NjY0"
MIME-Version: 1.0
Content-Length: 61
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Accept-Language: en-US,*
User-Agent: Mozilla/5.0
Host: krakonos.org

POST / HTTP/1.1
Content-Type: multipart/form-data; boundary="boundary_.oOo._OTI2NTYxNzU=MTk1MzY3MzMwNg==MTY3NjM1NjY0"
MIME-Version: 1.0
Content-Length: 61
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Accept-Language: en-US,*
User-Agent: Mozilla/5.0
Host: krakonos.org

--boundary_.oOo._OTI2NTYxNzU=MTk1MzY3MzMwNg==MTY3NjM1NjY0--
```

I have tried reproducing this problem with a simple example:

```
#include <QCoreApplication>
#include <QDebug>
#include <QNetworkAccessManager>
#include <QNetworkRequest>
#include <QHttpMultiPart>

int main(int argc, char **argv) {
    QCoreApplication app(argc, argv);

    QNetworkAccessManager man;

    QNetworkRequest req(QUrl("http://127.0.0.1:8001"));
    QHttpMultiPart mpt(QHttpMultiPart::FormDataType);
    man.post(req, &mpt);

    qDebug() << "hello world!";

    return app.exec();
}
```

Which works as expected. The mentioned behavior happens even if I create brand new QHttpMultiPart inside the HTTPRequest class, empty as used in the example.

However, the app is poorly written, contains a lot of duplicate code and is's very hard to understand. I would rather wait for the new API you advertised and write a new, simpler one.

I suspect the problem might disappear on some Linux distributions, as it might be mitigated by address space randomization or similar features.

File: OSV_Qt_upload/KQOAuth/libKQOAuth.prl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+QMAKE_PRL_BUILD_DIR = /home/mvexel/dev/upload-GUI/OSV_Qt_upload/KQOAuth
```
Comment (adrianb4-telenav) [inline OSV_Qt_upload/KQOAuth/libKQOAuth.prl:1]: this should be relative : $PWD/../KQOAuth

File: OSV_Qt_upload/UploadComponent/UploadComponent.pro [hunk 2]
```diff
@@ -98,11 +98,17 @@ DEPENDPATH += $$PWD/../zlib
 win32 {
    # LIBS+= $$OUT_PWD/../libusb/release/libusblib.lib
 }
-else:unix {
+else:macx {
     LIBS+= $$OUT_PWD/../libusb/libusblib.a
     LIBS += -framework CoreFoundation -framework Cocoa -framework IOKit
     INCLUDEPATH += $$PWD/../libusb/include/libusb-1.0
 }
+else:unix {
```
Comment (adrianb4-telenav) [inline OSV_Qt_upload/UploadComponent/UploadComponent.pro:106]: lib usb is not used anymore so after the code is merged i will remove this dependency

**Review by adrianb4-telenav — APPROVED** [review]
Overall looks nice, just some minor changes which i mentioned. Good job!

**Pull request merged**